modules/text_server_adv/text_server_adv.h

```diff
@@ -487,7 +491,7 @@ class TextServerAdvanced : public TextServerExtension {
 		/* Shaped data */
 		TextServer::Direction para_direction = DIRECTION_LTR; // Detected text direction.
 		int base_para_direction = UBIDI_DEFAULT_LTR;
-		bool valid = false; // String is shaped.
+		SafeFlag valid = SafeFlag(false); // String is shaped.
```

Comment:
```suggestion
		SafeFlag valid; // String is shaped.
```
In this case it's now unnecessary to assign